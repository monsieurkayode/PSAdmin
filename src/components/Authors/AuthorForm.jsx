import React from 'react';
import {
  shape, string, func, bool
} from 'prop-types';
import TextInput from '../common/TextInput/Index';

const AuthorForm = ({
  firstName,
  lastName,
  errors,
  onChange,
  onSubmit,
  onFocus,
  saving,
  loading
}) => (
  <form onSubmit={onSubmit}>
    <TextInput
      name="firstName"
      label="First Name"
      placeholder="Kent"
      onChange={onChange}
      onFocus={onFocus}
      loading={loading}
      value={firstName}
      error={errors.firstName}
    />
    <TextInput
      name="lastName"
      label="Last Name"
      placeholder="Dodds"
      onChange={onChange}
      onFocus={onFocus}
      loading={loading}
      value={lastName}
      error={errors.lastName}
    />
    <input
      type="submit"
      value={saving ? 'Saving...' : 'Save'}
      disabled={saving}
      className="btn btn-primary"
    />
  </form>
);

AuthorForm.defaultProps = {
  firstName: '',
  lastName: '',
  errors: {},
};

AuthorForm.propTypes = {
  firstName: string,
  lastName: string,
  errors: shape({}),
  saving: bool.isRequired,
  loading: bool.isRequired,
  onChange: func.isRequired,
  onSubmit: func.isRequired,
  onFocus: func.isRequired
};

export default AuthorForm;
