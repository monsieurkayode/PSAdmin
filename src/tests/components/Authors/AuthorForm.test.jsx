import React from 'react';
import { shallow } from 'enzyme';
import AuthorForm from '../../../components/Authors/AuthorForm';
import mockAuthor from '../../__mocks__/mockAuthor';

const setup = (saving) => {
  const props = {
    ...mockAuthor,
    loading: false,
    saving,
    onSubmit: jest.fn(),
    onFocus: jest.fn(),
    onChange: jest.fn()
  };

  const wrapper = shallow(<AuthorForm {...props} />);

  return {
    props,
    wrapper
  };
};

describe('AuthorForm Component', () => {
  const { wrapper, props } = setup(false);

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a form element', () => {
    expect(wrapper.find('form').length).toBe(1);
  });

  it('renders two TextInput children components', () => {
    expect(wrapper.find('TextInput').length).toBe(2);
  });

  it('renders a form submit button', () => {
    expect(wrapper.find('input[type="submit"]').length).toEqual(1);
  });

  it('form submit button displays Save text when not saving', () => {
    expect(wrapper.find('input[type="submit"]').props().value).toBe('Save');
  });

  it('form submit button displays Saving... text when saving', () => {
    const newWrapper = setup(true).wrapper;
    const saveButton = newWrapper.find('input[type="submit"]');
    expect(saveButton.props().value).toBe('Saving...');
  });

  it('calls onSumit when form is submitted', () => {
    wrapper.find('form').simulate('submit');
    expect(props.onSubmit).toHaveBeenCalled();
  });
});
