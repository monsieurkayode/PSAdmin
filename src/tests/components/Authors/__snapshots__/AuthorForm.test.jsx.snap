// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorForm Component renders correctly 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <TextInput
    error=""
    label="First Name"
    loading={false}
    name="firstName"
    onChange={[MockFunction]}
    onFocus={[MockFunction]}
    placeholder="Kent"
    value="Cory"
  />
  <TextInput
    error=""
    label="Last Name"
    loading={false}
    name="lastName"
    onChange={[MockFunction]}
    onFocus={[MockFunction]}
    placeholder="Dodds"
    value="House"
  />
  <input
    className="btn btn-primary"
    disabled={false}
    type="submit"
    value="Save"
  />
</form>
`;
